vscode-extensions.json

```diff
@@ -19,19 +20,26 @@
     },
     {
       "repository": "https://github.com/bmewburn/vscode-intelephense",
-      "revision": "v1.3.11"
+      "revision": "v1.3.11",
+      "dockerfile": "https://github.com/che-dockerfiles/che-sidecar-php",
```

Comment:
It's looking strange to have Dockerfile linking a repository no ?

Could we have more structure  like

```yaml
dockerImage :
   - name: quay.io/foo/bar
   - source:
       - url: github.com/foo/bar
       - revision:
       # if not at root /Dockerfile
       - path: ./subfolder/Dockerfile
```

Reply:
We don't need that level of detail right now, since we are maintaining all (but one of) the sidecar images they all have the tag format `version-SHA1 ID`. This allows us to do a simple check, if the `SHA1-ID` from the registry's sidecar doesn't match the `SHA1-ID` from the sidecar repo, then there is a mismatch and it's out of date.

Until we generalize (and enforce) a format for community contributed sidecars, we don't need to actually look at the Dockerfile.

Reply:
Field name is definitely wrong to me. And using same prefix for two vars means that a new level should be added

I would say that it's too fragile to rely on the fact that we maintain all images

Reply:
> Field name is definitely wrong to me. And using same prefix for two vars means that a new level should be added

Sure, a nested JSON object:
```js
{
  "repository": "https://github.com/bmewburn/vscode-intelephense",
  "revision": "v1.3.11",
  "dockerfile":{
    "repository": "https://github.com/che-dockerfiles/che-sidecar-php"
    "revision": "7"
    // optional
    "directory": "./subfolder/foo"
  }
}
```


> I would say that it's too fragile to rely on the fact that we maintain all images

Right now with the exception of one plugin, we maintain all sidecars. And for community contributors I am setting up repos in `che-dockerfiles` where they are using our build/naming system. So I'd say as an interim step we can use this and once we define a proper flow for community contributed sidecars, we can expand/change it. IMO it's better to get at least some automation now, we can always change it later.